canopen/pdo/base.py

```diff
@@ -22,10 +29,10 @@ class PdoBase(Mapping):
         Parent object associated with this PDO instance
     """
 
-    def __init__(self, node):
-        self.network = None
-        self.map = None  # instance of PdoMaps
-        self.node = node
+    def __init__(self, node: Union[LocalNode, RemoteNode]):
+        self.network: Optional[Network] = None
+        self.map: Optional[PdoMaps] = None
+        self.node: Union[LocalNode, RemoteNode] = node
```

Comment:
What is the intention of having `Union[LocalNode, RemoteNode]` as just using `BaseNode`? Do we expect other `BaseNode` derivatives that are not compatible with `PdoBase`

Reply:
Simply because we need the `.sdo` attribute here and that is not part of `BaseNode`.  I previously asked the same question in https://github.com/christiansandberg/canopen/pull/446#discussion_r1615969558.

Reply:
Yes, otherwise we only have network, object_dictionary and id attributes. Also explicit is better than implicit, thats why I don't like defining these attributes in the BaseNode which would be another solution.

Reply:
Great, thanks.

Perhaps we should at some point make a `NodeProtocol` for the required attributes instead of using a exhaustive list? It might not be worth the efforts thou, since the `Union` is only used a few places.

Reply:
Agreed, I think the `Union` approach is simple enough and warranted.  We can still rework it when there's a stronger argument for it.